src/NonLinearProgram/nlp_utilities.jl

```diff
@@ -0,0 +1,556 @@
+# Copyright (c) 2020: Andrew Rosemberg and contributors
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+"""
+    fill_off_diagonal(H)
+
+Filling the off-diagonal elements of a sparse matrix to make it symmetric.
+"""
+function fill_off_diagonal(H)
```

Comment:
Maybe constrain the input type? `H::SparseMatrixCSC`

Reply:
done!